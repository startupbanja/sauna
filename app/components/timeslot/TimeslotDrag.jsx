import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TimeslotDragable from './TimeslotDragable';

export const totalHeight = 400;

class TimeslotDrag extends Component {
  constructor(props) {
    super(props);
    this.handleAvailableChange = this.handleAvailableChange.bind(this);
  }

  handleAvailableChange(to, change) {
    this.props.onChange(to, (change / totalHeight) * (this.props.end - this.props.start));
  }

  render() {
    const containerStyle = {
      position: 'relative',
      height: totalHeight,
      overflow: 'hidden',
    };
    let starting = (this.props.available.start - this.props.start) * totalHeight;
    starting /= (this.props.end - this.props.start);
    let ending = (this.props.available.end - this.props.start) * totalHeight;
    ending /= (this.props.end - this.props.start);
    return (
      <div className="dragContainer" style={containerStyle}>
        <TimeslotDragable
          start={this.props.start}
          end={this.props.end}
          starting={0}
          ending={totalHeight}
          type="break"
          dragable={false}
          split={this.props.split}
          totalHeight={totalHeight}
        />
        <TimeslotDragable
          start={this.props.available.start}
          end={this.props.available.end}
          starting={starting}
          ending={ending}
          onChange={this.handleAvailableChange}
          type="available"
          dragable
        />
      </div>
    );
  }
}

TimeslotDrag.propTypes = {
  onChange: PropTypes.func.isRequired,
  end: PropTypes.number.isRequired,
  start: PropTypes.number.isRequired,
  available: PropTypes.shape({
    start: PropTypes.number,
    end: PropTypes.number,
  }).isRequired,
  split: PropTypes.number.isRequired,
};

export default TimeslotDrag;
