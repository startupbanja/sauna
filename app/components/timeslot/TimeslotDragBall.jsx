import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { totalHeight } from './TimeslotDrag';

const styles = {
  dragBall: {
    borderRadius: '100%',
    background: 'black',
    position: 'absolute',
    left: '50%',
  },
};

export class TimeslotDragBall extends Component {
  constructor(props) {
    super(props);
    Object.assign(styles.dragBall, { width: totalHeight * 0.03 });
    Object.assign(styles.dragBall, { height: totalHeight * 0.03 });
    this.handleDrag = this.handleDrag.bind(this);
    this.handleTouchMove = this.handleTouchMove.bind(this);
  }

  handleDrag(event) {
    if (event.clientY === 0) return;
    let origY = event.target.parentElement.offsetTop;
    origY += (event.target.offsetTop + (styles.dragBall.height / 2));
    const dragY = event.clientY - event.target.closest('.dragContainer').getBoundingClientRect().top;
    this.props.onChange(dragY - origY);
  }

  handleTouchMove(event) {
    if (event.targetTouches.item(0).clientY === 0) return;
    let origY = event.target.parentElement.offsetTop;
    origY += (event.target.offsetTop + (styles.dragBall.height / 2));
    const dragY = event.targetTouches.item(0).clientY - event.target.closest('.dragContainer').getBoundingClientRect().top;
    this.props.onChange(dragY - origY);
  }

  render() {
    let position = {};
    if (this.props.position === 'bottom') position = { bottom: -totalHeight * 0.015 };
    else if (this.props.position === 'top') position = { top: -totalHeight * 0.015 };
    return (
      <div
        draggable="true"
        onDrag={this.handleDrag}
        onTouchMove={this.handleTouchMove}
        style={Object.assign(position, styles.dragBall)}
      />
    );
  }
}

TimeslotDragBall.propTypes = {
  onChange: PropTypes.func.isRequired,
  position: PropTypes.oneOf(['top', 'bottom']).isRequired,
};
