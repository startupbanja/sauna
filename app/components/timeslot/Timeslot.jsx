import React from 'react';
import PropTypes from 'prop-types';
import TimeslotDrag from './TimeslotDrag';
import TimeslotInput from './TimeslotInput';

export function parseMinutes(timeString) {
  if (!timeString.match(/^([0-1]?\d|2[0-3]):[0-5]\d(:|$)/)) return false;
  const pieces = timeString.split(':');
  return (parseInt(pieces[0], 10) * 60) + parseInt(pieces[1], 10);
}
export function parseTimeStamp(minutes) {
  if (minutes < 0 || minutes >= 1440) return false;
  const hours = parseInt(minutes / 60, 10);
  let minutesOver = minutes % 60;
  if (minutesOver < 10) minutesOver = `0${minutesOver}`;
  return `${hours}:${minutesOver}`;
}

/* Component for presenting and editing users availabilities */
class Timeslot extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      available: {
        start: this.props.available.start,
        end: this.props.available.end,
      },
      start: this.props.start,
      end: this.props.end,
    };
    this.handleChange = this.handleChange.bind(this);
    this.submitAvailability = this.submitAvailability.bind(this);
    this.askForMoreTime = this.askForMoreTime.bind(this);
  }

  handleChange(to, change) {
    let newStart = this.state.available.start;
    let newEnd = this.state.available.end;
    if (to === 'start') {
      newStart = Math.min(Math.max(newStart + change, this.state.start), this.state.end);
      // newStart = Math.round(Math.min(newStart, newEnd));
      newEnd = Math.max(newStart, this.state.available.end);
    } else if (to === 'end') {
      newEnd = Math.max(Math.min(newEnd + change, this.state.end), this.state.start);
      // newEnd = Math.round(Math.max(newEnd, newStart));
      newStart = Math.min(newEnd, this.state.available.start);
    }
    const newObj = { available: { start: newStart, end: newEnd } };
    this.setState(newObj);
  }

  submitAvailability() {
    let startAvail = Math.round(this.state.available.start / 5) * 5;
    let endAvail = Math.round(this.state.available.end / 5) * 5;
    startAvail = this.askForMoreTime('start', startAvail);
    endAvail = this.askForMoreTime('end', endAvail);
    this.props.onSubmit(startAvail, endAvail);
  }

  askForMoreTime(type, availability) {
    if (type === 'start') {
      let start = Math.floor((availability - this.state.start) / this.props.split);
      start = (start * this.props.split) + this.state.start;
      const cond1 = availability - start < this.props.split / 2;
      const cond2 = availability - start > 0;
      if (cond1 && cond2) {
        if (confirm(`Could you come ${availability - start} minutes earlier?`)) { // eslint-disable-line
          this.setState({
            available: {
              start,
              end: this.state.available.end,
            },
          });
          return start;
        }
      }
    } else if (type === 'end') {
      let endTime = Math.ceil((availability - this.state.start) / this.props.split);
      endTime = (endTime * this.props.split) + this.state.start;
      const cond1 = endTime - availability < this.props.split / 2;
      const cond2 = endTime - availability > 0;
      if (cond1 && cond2) {
        if (confirm(`Could you stay ${endTime - availability} minutes longer?`)) { // eslint-disable-line
          this.setState({
            available: {
              start: this.state.available.start,
              end: endTime,
            },
          });
          return endTime;
        }
      }
    }
    return availability;
  }

  render() {
    const dateOptions = {
      weekday: 'short',
      day: 'numeric',
      month: 'numeric',
      year: 'numeric',
    };
    const moveToPrevClass = `glyphicon glyphicon-triangle-left${(this.props.onMoveToPrev === undefined) ? ' invisible' : ''}`;
    const moveToNextClass = `glyphicon glyphicon-triangle-right${(this.props.onMoveToNext === undefined) ? ' invisible' : ''}`;
    return (
      <div className="container timeslot-container">
        <p className="date">{this.props.date.toLocaleDateString('en-GB', dateOptions).replace(/\//g, '.')}</p>
        <p className="help-text help-text-center">
          Drag from the arrow to set your availability...
        </p>
        <TimeslotDrag
          start={this.state.start}
          end={this.state.end}
          available={this.state.available}
          onChange={this.handleChange}
          split={this.props.split}
        />
        <p className="help-text help-text-center">...or type in manually</p>
        <TimeslotInput
          available={this.state.available}
          onChange={this.handleChange}
        />
        <div className="navigation-container">
          <span
            className={moveToPrevClass}
            onClick={this.props.onMoveToPrev}
            role="button"
            tabIndex={0}
            onKeyDown={this.props.onMoveToPrev}
          />
          <button onClick={this.submitAvailability} className="btn btn-lg btn-major">Submit</button>
          <span
            className={moveToNextClass}
            onClick={this.props.onMoveToNext}
            role="button"
            tabIndex={0}
            onKeyDown={this.props.onMoveToNext}
          />
        </div>
      </div>
    );
  }
}

Timeslot.propTypes = {
  date: PropTypes.objectOf(Date).isRequired,
  start: PropTypes.number.isRequired,
  end: PropTypes.number.isRequired,
  available: PropTypes.shape({
    start: PropTypes.number.isRequired,
    end: PropTypes.number.isRequired,
  }).isRequired,
  split: PropTypes.number.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onMoveToPrev: PropTypes.func,
  onMoveToNext: PropTypes.func,
};

Timeslot.defaultProps = {
  onMoveToNext: undefined,
  onMoveToPrev: undefined,
};

export default Timeslot;
