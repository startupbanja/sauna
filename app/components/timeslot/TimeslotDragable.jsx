import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TimeslotDragBall } from './TimeslotDragBall';
import { parseTimeStamp } from './Timeslot';

const styles = {
  time: {
    position: 'absolute',
    right: '5px',
    margin: '0px',
  },
};
export class TimeslotDragable extends Component {
  constructor(props) {
    super(props);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.handleStartChange = this.handleStartChange.bind(this);
  }

  handleStartChange(change) {
    this.props.onChange('start', change);
  }
  handleEndChange(change) {
    this.props.onChange('end', change);
  }

  render() {
    const containerStyle = {
      background: 'orange',
      position: 'absolute',
      height: this.props.ending - this.props.starting,
      top: this.props.starting,
      width: '100%',
    };
    if (this.props.dragable !== true) containerStyle.background = 'gray';
    else containerStyle.background = 'orange';
    let topDragBall = null;
    let bottomDragBall = null;
    if (this.props.dragable === true) {
      topDragBall = <TimeslotDragBall position="top" onChange={this.handleStartChange} />;
      bottomDragBall = <TimeslotDragBall position="bottom" onChange={this.handleEndChange} />;
    }
    return (
      <div className="dragableContainer" style={containerStyle}>
        <p style={Object.assign({ top: '5px' }, styles.time)}>{parseTimeStamp(this.props.start)}</p>
        <p style={Object.assign({ bottom: '5px' }, styles.time)}>{parseTimeStamp(this.props.end)}</p>
        {topDragBall}
        {bottomDragBall}
      </div>
    );
  }
}

TimeslotDragable.propTypes = {
  onChange: PropTypes.func,
  ending: PropTypes.number.isRequired,
  starting: PropTypes.number.isRequired,
  dragable: PropTypes.bool.isRequired,
  start: PropTypes.number.isRequired,
  end: PropTypes.number.isRequired,
};
