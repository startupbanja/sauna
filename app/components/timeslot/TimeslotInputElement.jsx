import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { parseTimeStamp, parseMinutes } from './Timeslot';

const styles = {
  input: {
    position: 'absolute',
    right: 0,
  },
};

function handleKeypress(event) {
  if (event.key === 'Enter') {
    event.target.blur();
  }
}

export class TimeslotInputElement extends Component {
  constructor(props) {
    super(props);
    this.state = { editedValue: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleChange(event) {
    this.setState({ editedValue: event.target.value });
  }
  handleBlur(event) {
    this.setState({ editedValue: '' });
    if (parseMinutes(event.target.value) !== false) {
      this.props.onChange(parseMinutes(event.target.value) - this.props.time);
    }
  }

  render() {
    let value = parseTimeStamp(this.props.time);
    if (this.state.editedValue !== '') value = this.state.editedValue;
    return (
      <input
        onKeyPress={handleKeypress}
        onChange={this.handleChange}
        onBlur={this.handleBlur}
        style={styles.input}
        value={value}
        type="text"
      />
    );
  }
}

TimeslotInputElement.propTypes = {
  onChange: PropTypes.func.isRequired,
  time: PropTypes.number.isRequired,
};
