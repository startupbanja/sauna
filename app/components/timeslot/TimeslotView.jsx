import React, { Component } from 'react';
import pageContents from '../pageContent';
import Timeslot, { parseMinutes, parseTimeStamp } from './Timeslot';

/* Component to handle availability data between backend and Timeslot */
class TimeslotView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      data: [],
    };
    this.renderTimeslot = this.renderTimeslot.bind(this);
    this.fetchData = this.fetchData.bind(this);
    this.submitAvailability = this.submitAvailability.bind(this);
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    pageContents.fetchData('/getComingMeetingDays', 'GET', {})
      .then((result) => {
        const newData = [];
        result.forEach((day) => {
          const start = parseMinutes(new Date(`${day.date}T${day.startTime}`).toTimeString().substr(0, 5));
          const end = parseMinutes(new Date(`${day.date}T${day.endTime}`).toTimeString().substr(0, 5));
          newData.push({
            date: new Date(day.date),
            start,
            end,
            split: day.split,
            available: {
              start: (day.time === null) ? start : parseMinutes(day.time),
              end: (day.duration === null) ? end : parseMinutes(day.time) + day.duration,
            },
          });
        });
        this.setState({ data: newData });
      });
  }

  submitAvailability(startAvail, endAvail) {
    const meetingDay = this.state.data[this.state.index];
    let startTime = Math.ceil((startAvail - meetingDay.start) / meetingDay.split);
    startTime = (startTime * meetingDay.split) + meetingDay.start;
    let endTime = Math.floor((endAvail - meetingDay.start) / meetingDay.split);
    endTime = (endTime * meetingDay.split) + meetingDay.start;
    pageContents.fetchData('/insertAvailability', 'POST', {
      date: meetingDay.date.toISOString().substr(0, 10),
      start: parseTimeStamp(startTime),
      end: parseTimeStamp(endTime),
    }).then((result) => {
      if (result.status === 'success') {
        const oldData = this.state.data;
        oldData[this.state.index].available = {
          start: startTime,
          end: endTime,
        };
        this.setState({
          data: oldData,
        });
        this.changeDate(1);
      }
    });
  }

  changeDate(diff) {
    this.setState({
      index: Math.min(this.state.data.length - 1, Math.max(0, this.state.index + diff)),
    });
  }

  renderTimeslot() {
    if (this.state.data.length > 0) {
      const day = this.state.data[this.state.index];
      return (
        <Timeslot
          key={day.date}
          start={day.start}
          end={day.end}
          split={day.split}
          available={day.available}
          date={day.date}
          onSubmit={this.submitAvailability}
          onMoveToPrev={((this.state.index > 0) || undefined) && (() => this.changeDate(-1))}
          onMoveToNext={((this.state.index < this.state.data.length - 1) || undefined)
            && (() => this.changeDate(1))}
        />
      );
    }
    return undefined;
  }

  render() {
    return (
      <div className="timeslot-picker">
        <link rel="stylesheet" type="text/css" href="app/styles/timeslot_style.css" />
        {this.renderTimeslot()}

      </div>
    );
  }
}

export default TimeslotView;
