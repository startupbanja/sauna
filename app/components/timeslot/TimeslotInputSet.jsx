import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TimeslotInputElement from './TimeslotInputElement';
/* eslint jsx-a11y/label-has-for: "warn"
*/

class TimeslotInputSet extends Component {
  constructor(props) {
    super(props);
    this.handleStartChange = this.handleStartChange.bind(this);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.isUnavailable = this.isUnavailable.bind(this);
    this.markAsUnavailable = this.markAsUnavailable.bind(this);
    this.state = {
      errorIn: '',
    };
  }

  handleStartChange(change) {
    this.props.onChange('start', change);
  }
  handleEndChange(change) {
    this.props.onChange('end', change);
  }

  isUnavailable() {
    return this.props.start === this.props.end;
  }
  markAsUnavailable(type) {
    if (type === 'start') {
      this.props.onChange('start', this.props.end - this.props.start);
      this.setState({ errorIn: 'start' });
    } else if (type === 'end') {
      this.props.onChange('end', this.props.start - this.props.end);
      this.setState({ errorIn: 'end' });
    }
  }

  render() {
    return (
      <div className="timeslot-input">
        <TimeslotInputElement
          id="start"
          time={this.props.start}
          onChange={this.handleStartChange}
          markAsUnavailable={() => this.markAsUnavailable('start')}
          unavailable={this.isUnavailable() && (this.state.errorIn === 'start')}
        />
        <p>-</p>
        <TimeslotInputElement
          time={this.props.end}
          onChange={this.handleEndChange}
          unavailable={this.isUnavailable() && (this.state.errorIn !== 'start')}
          markAsUnavailable={() => this.markAsUnavailable('end')}
        />
        <p className={`unavailable-text${(!this.isUnavailable()) ? ' invisible' : ''}`}>
          unavailable
        </p>
      </div>
    );
  }
}

TimeslotInputSet.propTypes = {
  onChange: PropTypes.func.isRequired,
  start: PropTypes.number.isRequired,
  end: PropTypes.number.isRequired,
};

export default TimeslotInputSet;
