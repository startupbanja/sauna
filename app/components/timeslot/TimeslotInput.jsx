import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TimeslotInputSet } from './TimeslotInputSet';

export class TimeslotInput extends Component {
  render() {
    return (
      <div>
        <TimeslotInputSet
          start={this.props.available.start}
          end={this.props.available.end}
          type="available"
          onChange={this.props.onChange}
        />
      </div>
    );
  }
}

TimeslotInput.propTypes = {
  available: PropTypes.shape({
    start: PropTypes.number.isRequired,
    end: PropTypes.number.isRequired,
  }).isRequired,
  onChange: PropTypes.func.isRequired,
};
