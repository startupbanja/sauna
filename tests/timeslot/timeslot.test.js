import React from 'react';
import {parseMinutes, parseTimeStamp, Timeslot} from '../../app/components/timeslot/Timeslot';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('parseMinutes works correctly', () => {
    expect(parseMinutes("string")).toBe(false);
    expect(parseMinutes("24:00")).toBe(false);
    expect(parseMinutes("10")).toBe(false);
    expect(parseMinutes("3:63")).toBe(false);
    expect(parseMinutes("9:00")).toBe(540);
    expect(parseMinutes("13:24")).toBe(804);
    expect(parseMinutes("-4:00")).toBe(false);
    expect(parseMinutes("23:59")).toBe(1439);
    expect(parseMinutes("0:00")).toBe(0);
});

test('parseTimeStamp works correctly', () => {
    expect(parseTimeStamp(-1)).toBe(false);
    expect(parseTimeStamp(1440)).toBe(false);
    expect(parseTimeStamp(0)).toBe("0:00");
    expect(parseTimeStamp(512)).toBe("8:32");
    expect(parseTimeStamp(770)).toBe("12:50");
    expect(parseTimeStamp(1439)).toBe("23:59");
    expect(parseTimeStamp(600)).toBe("10:00");
});

test('Timeslot initializes correctly', () => {
    var timeslot = renderer.create(<Timeslot start="8:00" end="12:00" />);
    let tree = timeslot.toJSON();
    expect(tree).toMatchSnapshot();
    timeslot = renderer.create(<Timeslot start="12:20" end="20:40"/>);
    tree = timeslot.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Timeslot handles change correctly', () => {
    const timeslot = shallow(<Timeslot start="10:00" end="12:00" />);
    timeslot.instance().handleChange('start', -30);
    expect(timeslot.state().available).toEqual({start: 600, end: 600});
    timeslot.instance().handleChange('end', 130);
    expect(timeslot.state().available).toEqual({start: 600, end: 720});
    timeslot.instance().handleChange('start', 10);
    expect(timeslot.state().available).toEqual({start: 610, end: 720});
    timeslot.instance().handleChange("end", 20);
    expect(timeslot.state().available).toEqual({start: 610, end: 720});
    timeslot.instance().handleChange("end", -35);
    expect(timeslot.state().available).toEqual({start: 610, end: 685});
    expect(timeslot).toMatchSnapshot();
    timeslot.instance().handleChange("end", 26);
    expect(timeslot.state().available).toEqual({start: 610, end: 711});
    timeslot.instance().handleChange("start", 130);
    expect(timeslot.state().available).toEqual({start: 711, end: 711});
    expect(timeslot).toMatchSnapshot();
});