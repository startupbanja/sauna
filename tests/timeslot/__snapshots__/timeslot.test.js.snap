// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeslot handles change correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeslot
    end="12:00"
    start="10:00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TimeslotDrag
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          end={720}
          onChange={[Function]}
          start={600}
        />,
        <TimeslotInput
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          onChange={[Function]}
        />,
      ],
      "style": Object {
        "width": "300px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "end": 720,
          "onChange": [Function],
          "start": 600,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TimeslotDrag
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            end={720}
            onChange={[Function]}
            start={600}
          />,
          <TimeslotInput
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            onChange={[Function]}
          />,
        ],
        "style": Object {
          "width": "300px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "end": 720,
            "onChange": [Function],
            "start": 600,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Timeslot handles change correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeslot
    end="12:00"
    start="10:00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TimeslotDrag
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          end={720}
          onChange={[Function]}
          start={600}
        />,
        <TimeslotInput
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          onChange={[Function]}
        />,
      ],
      "style": Object {
        "width": "300px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "end": 720,
          "onChange": [Function],
          "start": 600,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TimeslotDrag
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            end={720}
            onChange={[Function]}
            start={600}
          />,
          <TimeslotInput
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            onChange={[Function]}
          />,
        ],
        "style": Object {
          "width": "300px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "end": 720,
            "onChange": [Function],
            "start": 600,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Timeslot initializes correctly 1`] = `
<div
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="dragContainer"
    style={
      Object {
        "height": 400,
        "position": "relative",
      }
    }
  >
    <div
      className="dragableContainer"
      style={
        Object {
          "background": "gray",
          "height": 400,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <p
        style={
          Object {
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
            "top": "5px",
          }
        }
      >
        8:00
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
          }
        }
      >
        12:00
      </p>
    </div>
    <div
      className="dragableContainer"
      style={
        Object {
          "background": "orange",
          "height": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <p
        style={
          Object {
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
            "top": "5px",
          }
        }
      >
        8:00
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
          }
        }
      >
        8:00
      </p>
      <div
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "background": "black",
            "borderRadius": "100%",
            "height": 12,
            "left": "50%",
            "position": "absolute",
            "top": -6,
            "width": 12,
          }
        }
      />
      <div
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "background": "black",
            "borderRadius": "100%",
            "bottom": -6,
            "height": 12,
            "left": "50%",
            "position": "absolute",
            "width": 12,
          }
        }
      />
    </div>
  </div>
  <div>
    <div>
      <div
        style={
          Object {
            "margin": "5px",
            "position": "relative",
          }
        }
      >
        <label>
          Start
        </label>
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
          type="text"
          value="8:00"
        />
      </div>
      <div
        style={
          Object {
            "margin": "5px",
            "position": "relative",
          }
        }
      >
        <label>
          End
        </label>
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
          type="text"
          value="8:00"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Timeslot initializes correctly 2`] = `
<div
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="dragContainer"
    style={
      Object {
        "height": 400,
        "position": "relative",
      }
    }
  >
    <div
      className="dragableContainer"
      style={
        Object {
          "background": "gray",
          "height": 400,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <p
        style={
          Object {
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
            "top": "5px",
          }
        }
      >
        12:20
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
          }
        }
      >
        20:40
      </p>
    </div>
    <div
      className="dragableContainer"
      style={
        Object {
          "background": "orange",
          "height": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <p
        style={
          Object {
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
            "top": "5px",
          }
        }
      >
        12:20
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
            "margin": "0px",
            "position": "absolute",
            "right": "5px",
          }
        }
      >
        12:20
      </p>
      <div
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "background": "black",
            "borderRadius": "100%",
            "height": 12,
            "left": "50%",
            "position": "absolute",
            "top": -6,
            "width": 12,
          }
        }
      />
      <div
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "background": "black",
            "borderRadius": "100%",
            "bottom": -6,
            "height": 12,
            "left": "50%",
            "position": "absolute",
            "width": 12,
          }
        }
      />
    </div>
  </div>
  <div>
    <div>
      <div
        style={
          Object {
            "margin": "5px",
            "position": "relative",
          }
        }
      >
        <label>
          Start
        </label>
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
          type="text"
          value="12:20"
        />
      </div>
      <div
        style={
          Object {
            "margin": "5px",
            "position": "relative",
          }
        }
      >
        <label>
          End
        </label>
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
          type="text"
          value="12:20"
        />
      </div>
    </div>
  </div>
</div>
`;
